### PR TITLE
HTTP Log Tool and More Script Automations

### PR DESCRIPTION
**Pull request by emil916**
- added new HTTP Perf Log mechanism 
   (It measures http session lifetime, receive and send block durations)
    - automated generation of spec.json
    - combined HEY and LOADTEST scripts into one, optional

### CODE REVIEW

File: runtime/include/http_session.h [hunk 1]
```diff
@@ -54,9 +54,13 @@ struct http_session {
 	size_t                  response_buffer_written;
 	struct tenant          *tenant; /* Backlink required when read blocks on listener core */
 	uint64_t                request_arrival_timestamp;
+	uint64_t                request_downloaded_timestamp;
+	uint64_t                response_takeoff_timestamp;
```
Comment (bushidocodes) [inline runtime/include/http_session.h:58]: What does takeoff mean in this context?

Comment (emil916) [reply runtime/include/http_session.h:58]: It means response sending started. The naming might not be very successful. But I am measuring request receival and response sending separately. So, each has begin and end timestamps.

File: tests/bash_libraries/generate_spec_json.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# shellcheck shell=bash
+# shellcheck disable=SC2154
+if [ -n "$__generate_spec_json_sh__" ]; then return; fi
+__generate_spec_json_sh__=$(date)
+
+
+generate_spec_json() {
```
Comment (bushidocodes) [inline tests/bash_libraries/generate_spec_json.sh:7]: Looks pretty useful!

**Review by bushidocodes — APPROVED** [review]
LGTM

**Pull request merged**